### ISSUE TITLE
Error in saturation_pressure calculation

### ISSUE THREAD

**Issue by biglimp** [ladybug-tools/uwg#262]
I am experiencing an error where a log of zero or a negative value probably happens. Input files are attached. 

Version of uwg according to `pip list` is 5.8.7 (using uwg.__version__ gives me 5.4.3, maybe needs to be updated?). 

The region is Gothenburg, Sweden and I have successfully run uwg on similar settings (also attached). I also been able to run this in previous versions of uwg.

```
model.simulate()
File "C:\OSGeo4W\apps\Python39\lib\site-packages\uwg\uwg.py", line 1350, in simulate
self.UCM, self.UBL, self.BEM = urbflux(
File "C:\OSGeo4W\apps\Python39\lib\site-packages\uwg\urbflux.py", line 38, in urbflux
BEM[j].building.BEMCalc(UCM, BEM[j], forc, parameter, simTime)
File "C:\OSGeo4W\apps\Python39\lib\site-packages\uwg\building.py", line 539, in BEMCalc
psychrometrics(self.indoor_temp, self.indoor_hum, forc.pres)
File "C:\OSGeo4W\apps\Python39\lib\site-packages\uwg\psychrometrics.py", line 41, in psychrometrics
Pws = saturation_pressure(Tdb) # Get saturation pressure for given Tdb
File "C:\OSGeo4W\apps\Python39\lib\site-packages\uwg\psychrometrics.py", line 77, in saturation_pressure
6.5459673 * log(T)) # Pa
ValueError: math domain error
```
[error.zip](https://github.com/ladybug-tools/uwg/files/7580072/error.zip)
[success.zip](https://github.com/ladybug-tools/uwg/files/7580073/success.zip)

**Comment by saeranv** [comment]
@biglimp are you making a custom building type?

**Comment by biglimp** [comment]
No, I am using the existing ones from the pickle coming with the pypi downlaod.

**Comment by saeranv** [comment]
This is something that we've run into before (there's a note in the code about it). The problem may be due to large timesteps (i.e. 3600s) causing instability in the numerical differentiation. Is your timestep very large, and if so, can you try rerunning with something smaller, like 300s? 

And I agree, best to update UWG to its latest version.

**Comment by biglimp** [comment]
Timesteps (dtSim) is already in 300s. I also tried to change the building type to MidRiseApartments but still the same error.

**Comment by biglimp** [comment]
Some additional information. I played around a bit and found that if I increased the verToHor parameter I could run the files in error.zip without error. The error area has a few large footprint buildings which might cause the model to fail...

**Comment by biglimp** [comment]
Any updates on this issue?

I debugged the code a bit more using the input in the attached _error.zip_ and found out that it is the **quickservicerestaurant** building class that is causing the issue. This class is unstable using low **verToHor** numbers creating **T_wallex** and **T_wallin** to become very large negative numbers, which in turn finally cause the saturation pressure calculation too fail. 

Why is building classes with no fractions in the uwg-file included in the calculation in the first place?

**Comment by PetitMote** [comment]
Hi,
Just to say I ran into this error too. Increasing verToHor to 0.2 was enough to proceed to the calculation, but it’s a poor way to solve the issue.
